src/ophyd_async/core/_readable.py

```diff
@@ -150,19 +150,19 @@ def add_children_as_readables(
         :meth:`HintedSignal.uncached`
         """
 
-        dict_copy = self.__dict__.copy()
+        dict_copy = dict(self.children())
```

Comment:
May as well only take the keys we need
```suggestion
        keys_before = set(self.children().keys())
```
...

Reply:
`self.children()` is not a dict yet, so we need to turn it into one first...